PWGHF/vertexingHF/AliAnaTaskSEXic0SL.C

```diff
@@ -0,0 +1,947 @@
+			//LHC16k and LHC16l are CD dedicated runs! CD-online-trigger is used
+			if ( (fEvt->GetFiredTriggerClasses().Contains("CINT7-B-NOPF-CENT")) &&
+				 (fTaskOpt.Contains("LHC16k") || fTaskOpt.Contains("LHC16l")) &&
+				 (TrigStore[a] == AliVEvent::kINT7) ) t_trigFired = true;
+		}
+		if (!t_trigFired) return; //Return 7d
+		else fHisto->FillTH1("EvtCut", "Trig>0", 1); //@
+		const bool t_trigFiredMB   = (fEvtTrig & AliVEvent::kINT7)?true:false;
+		const bool t_trigFiredHMV0 = (fEvtTrig & AliVEvent::kHighMultV0)?true:false;
+
+		fHisto->FillTH1("EvtTrig", "Any", 1); //@
+		if (t_trigFiredMB) fHisto->FillTH1("EvtTrig", "MB", 1); //@
+		if (t_trigFiredHMV0) fHisto->FillTH1("EvtTrig", "HMV0", 1); //@
+
+		//Activate RDHF cut by invoking it, then check pileup status by using multi vertexer
+		const bool t_evtGoodMB   = fEvtCutsMB  ->IsEventSelected(fEvt);
+		const bool t_evtGoodHMV0 = fEvtCutsHMV0->IsEventSelected(fEvt);
+		fEvtPileupMB   = fEvtCutsMB  ->IsEventRejectedDueToPileup();
+		fEvtPileupHMV0 = fEvtCutsHMV0->IsEventRejectedDueToPileup();
+		if (fEvtPileupMB && fEvtPileupHMV0) return; //Return 8d
+		else fHisto->FillTH1("EvtCut", "PileupX>0", 1); //@
+
+		//Check vtx validity (NOT vtxZ)
+		fEvtVtx = fEvt->GetPrimaryVertex();
+		if (!fEvtVtx || fEvtVtx->GetNContributors()<cut_vtxNContributors) return; //Return 9d
+
+		//Store AliNormalizationCounter: DO NOT CHANGE THIS POSITION! (after vtx found, before |VtxZ| < 10)
+		if (t_trigFiredMB)
+		{
+			if (fEvtMult >= 0.0 && fEvtMult <= 100.0)
+			{
+				fANC_MB_0to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+				if (fEvtINELLgt0) fANC_INEL0_MB_0to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+			}
+			if (fEvtMult >= 30.0 && fEvtMult <= 100.0)
+			{
+				fANC_MB_30to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+				if (fEvtINELLgt0) fANC_INEL0_MB_30to100->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+			}
+			if (fEvtMult >= 0.1 && fEvtMult <= 30.0)
+			{
+				fANC_MB_0p1to30->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+				if (fEvtINELLgt0) fANC_INEL0_MB_0p1to30->StoreEvent(fEvt, fEvtCutsMB, IsMC);
+			}
+		}
+		if (t_trigFiredHMV0 && fEvtMult >= 0.0 && fEvtMult <= 0.1)
+		{
+			fANC_HMV0_0to0p1->StoreEvent(fEvt, fEvtCutsHMV0, IsMC);
+			if (fEvtINELLgt0) fANC_INEL0_HMV0_0to0p1->StoreEvent(fEvt, fEvtCutsHMV0, IsMC);
+		}
+
+		//Check vtxZ
+		fEvtVtxZ = fEvtVtx->GetZ();
+		if (fabs(fEvtVtxZ) > cut_vtxZ) return; //Return 10d
+		else fHisto->FillTH1("EvtCut", "VtxZ10", 1); //@
+
+		//Check INEL>0
+		if (AliPPVsMultUtils::IsINELgtZERO(fEvt)==true)	fEvtINELLgt0 = true;
+		if (fEvtINELLgt0) fHisto->FillTH1("EvtCut", "INEL>0", 1); //@
+	}//data
+	else //MC
+	{
+		fMCEvt = fInputHandler->MCEvent();
+
+		//Check vtxZ
+		fEvtVtx = fMCEvt->GetPrimaryVertex();
+		fEvtVtxZ = fEvtVtx->GetZ();
+		if (fabs(fEvtVtxZ) > cut_vtxZ) return; //Return 7mc
+
+		//Get true Xic0
+		int nXic0 = 0;
+		const int nTracksMC = fMCEvt->GetNumberOfTracks();
+		for (int a=0; a<nTracksMC; a++)
+		{
+			fMCPart = (AliAODMCParticle*)fMCEvt->GetTrack(a);
```

Comment:
This is not the same as storing all generated Xic (i.e. incorrect for the efficiency and acceptance correction). You need to loop over the MC generated TClonesArray instead. See for example: https://github.com/alisw/AliPhysics/blob/master/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEHFTreeCreatorApply.cxx#L1710

Reply:
It is also not the same as storing reconstructed Xic (but I don't think that was your aim here). For the proper efficiency correction you need to store both the generated (as above) and the reconstructed+selected eXi pairs that are matched to a Xic at MC truth level

Reply:
1. The functions I use are:

a. AliMCEvent->GetNumberOfTracks() ([link] http://alidoc.cern.ch/AliRoot/v5-09-24/_ali_m_c_event_8h_source.html#l00098)) 
`virtual Int_t GetNumberOfTracks() const {return fNparticles;}`

If you check the function AliMCEvent::SetParticleArray(TClonesArray* mcParticles) ([link](http://alidoc.cern.ch/AliRoot/v5-09-24/_ali_m_c_event_8cxx_source.htm )),
```
fMCParticles = mcParticles;
fNparticles = fMCParticles->GetEntries();
```

b. AliMCEvent->GetTrack();
`virtual AliVParticle *GetTrack(Int_t i) const;

2. The functions in your reference are:

```
 for(Int_t iPart=0; iPart<arrayMC->GetEntriesFast(); iPart++){
    
    AliAODMCParticle* mcPart = dynamic_cast<AliAODMCParticle*>(arrayMC->At(iPart));
```

In conclusion, they look basically the same.